nefertari/scripts/es.py

```diff
@@ -1,138 +1,272 @@
+        log.info('Total indexed tems count {} for model {}'.format(len(set(initial_dict[model_name])), model_name))
+
+        assert len(initial_dict[model_name]) == len(set(initial_dict[model_name]))
+        assert len(initial_dict[model_name]) == items_count
+
+
+def get_logger():
+    log = logging.getLogger()
+    log.setLevel(logging.WARNING)
+    ch = logging.StreamHandler()
+    formatter = logging.Formatter('%(asctime)s - %(name)s - %(message)s')
+    ch.setFormatter(formatter)
+    log.addHandler(ch)
+    return log
+
 
-        # Prevent ES.setup_mappings running on bootstrap;
-        # Restore ES._mappings_setup after bootstrap is over
-        mappings_setup = getattr(ES, '_mappings_setup', False)
-        try:
-            ES._mappings_setup = True
-            env = self.bootstrap[0](self.options.config)
-        finally:
-            ES._mappings_setup = mappings_setup
+def setup_app(options, put_mappings):
+    # Prevent ES.setup_mappings running on bootstrap;
+    # Restore ES._mappings_setup after bootstrap is over
+    log = get_logger()
+    mappings_setup = getattr(ES, '_mappings_setup', False)
+    try:
+        ES._mappings_setup = True
+        env = bootstrap(options.config)
+    finally:
+        ES._mappings_setup = mappings_setup
+    registry = env['registry']
+    # Include 'nefertari.engine' to setup specific engine
+    config = Configurator(settings=registry.settings)
+    config.include('nefertari.engine')
+    ES.setup(dictset(registry.settings))
 
-        registry = env['registry']
-        # Include 'nefertari.engine' to setup specific engine
-        config = Configurator(settings=registry.settings)
-        config.include('nefertari.engine')
+    if put_mappings:
+        log.setLevel(logging.INFO)
+        ES.setup_mappings()
+    return registry
 
-        self.log = log
+def recreate_index(registry):
+    log = get_logger()
+    settings = dictset(registry.settings)
+    log.info('Deleting index')
+    ES.delete_index()
+    log.info('Creating index')
+    create_index_with_settings(settings)
+    log.info('Creating mappings')
+    ES.setup_mappings()
 
-        if not self.options.quiet:
-            self.log.setLevel(logging.INFO)
 
-        self.settings = dictset(registry.settings)
+class ESTasksPool(object):
 
-    def index_models(self, model_names):
-        self.log.info('Indexing models documents')
-        params = self.options.params or ''
-        params = dict([[k, v[0]] for k, v in urllib.parse.parse_qs(params).items()])
+    def __init__(self, model_names, processes, options):
+        self._pool = {process_id: [] for process_id in range(0, processes)}
+        self.options = options
 
         for model_name in model_names:
-            self.log.info('Processing model `{}`'.format(model_name))
+            for process_id in range(0, processes):
+                self._pool[process_id].append(ESTask(model_name, options))
```

Comment:
Wouldn't it be simpler and more readable to reverse the `for` loop nesting and do something like this:
```python
self._pool = {}
for process_id in range(0, processes):
    self._pool[process_id] = []
    for model_name in model_names:
        self._pool[process_id].append(ESTask(model_name, options))
```